chore: improve cucumber tags and run time speed

SWvheerden opened the pull request:
Description
---

This PR improves the run time of certain tests to remove `waits` and make sure they have async functions to retest waitable events.
It adds a `@base-node` tag to all base node tests.
It adds a `@wallet` tag to all wallet tests.
It adds `@`  tags to all feature files.
Its adds and remove `@critical` tags to all required tests, adding where critical functionality is not tested and removing where it is double tested and or not critical to core functionality.

Motivation and Context
---
Made sure we can run all tests as short as possible. Tests are grouped with the following tags:
`@base-node` which can be used to run tests if verification of code base node functionality is desired
`@wallet` which can be used to run tests if verification of code wallet functionality is desired
`@critical` if a short test of all core functionality is desired.
`@long-running` if really detailed long-running tests are desired. Can take hours. 

Tests without a `@critical` and `@long-running` are tests that are required for passing CI, but will take longer to run.
Running test with `--tags @base-node AND @critical` or `--tags @wallet AND @critical` completes it run time below 10 minutes.

stringhandler commented on integration_tests/features/WalletQuery.feature:
```diff
@@ -1,13 +1,14 @@
-@wallet
+@wallet-query
```
```suggestion
@wallet-query @wallet
```

stringhandler commented on integration_tests/features/WalletMonitoring.feature:
```diff
@@ -51,7 +48,7 @@ Feature: Wallet Monitoring
 
   # 18+ mins on circle ci
-  @long-running
+  @long-running @wallet
```
You can just put this tag at the top of the file

SWvheerden replied:
Moved all '@wallet' and all '@base-node' tags where I could to the top

philipr-za reviewed (commented):
The CI run of the integration tests looks pretty broken?

delta1 commented:
Not sure what’s gone wrong with the grpc client now

SWvheerden committed:
improve cucumber tags and speed

Add flaky and broken tags
update readme
shorten tests and move more to long-running

SWvheerden committed:
move @base_node and @wallet tags to top of files

philipr-za reviewed (approved):
LGTM
